docs/tasks-rbac-permissions.md

```diff
@@ -0,0 +1,140 @@
+The name of the service account is usually named `${TASK_NAME}-task`.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: my-run
+spec:
+  pipelineRef:
+    name: my-pipeline
+  serviceAccountNames:
+    - serviceAccountName: create-vm-from-manifest-task
+      taskName: create-vm-from-manifest
+    - serviceAccountName: execute-in-vm-task
+      taskName: execute-in-vm
+
+```
+
+## Enhanced RBAC permissions
+
+The default deployment of KubeVirt Tekton Tasks is initialized with permissions (`RoleBinding`) for only a specific namespace.
+By default, it is not possible for example to create a pipeline in namespace `ns-a` that would fetch/create resources (eg `VirtualMachines`) in namespace `ns-b`.
+To support such use cases additional bindings need to be created.
+
+### Multi Namespace RBAC permissions
+
+Let's assume the tasks and thus their service accounts are deployed in `ns-a`.
+To allow access to `ns-b`, the following `RoleBinding` should be created for each task.
+Only two tasks are used for simplicity in the following example.
+
+```bash
+#!/usr/bin/env bash
+for TASK_NAME in generate-ssh-keys create-vm-from-manifest; do
+    kubectl apply -f - << EOF
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: ${TASK_NAME}-task
+  namespace: ns-b
+roleRef:
+  kind: ClusterRole
+  name: ${TASK_NAME}-task
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+  - kind: ServiceAccount
+    name:  ${TASK_NAME}-task
+    namespace: ns-a
+EOF
+done
+```
+
+This example enables `generate-ssh-keys` and `create-vm-from-manifest` tasks to create secrets/VMs in `ns-b` from a pipeline started in `ns-a`.
+
+Warning: this will allow users with access to `generate-ssh-keys-task` and `create-vm-from-manifest-task` service accounts to run pods which can access `ns-b`.
+
+
+### Cluster RBAC permissions
+
+Let's assume the tasks and thus their service accounts are deployed in `ns-a`.
+To allow access to any other namespace, the following `ClusterRoleBinding` should be created for each task.
+Only two tasks are used for simplicity in the following example.
+
+```bash
+#!/usr/bin/env bash
+for TASK_NAME in generate-ssh-keys create-vm-from-manifest; do
+    kubectl apply -f - << EOF
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: ${TASK_NAME}-task
+roleRef:
+  kind: ClusterRole
+  name: ${TASK_NAME}-task
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+  - kind: ServiceAccount
+    name:  ${TASK_NAME}-task
+    namespace: ns-a
+EOF
+done
+```
+
+This example enables `generate-ssh-keys` and `create-vm-from-manifest` tasks to create secrets/VMs in any namespace in the cluster from a pipeline started in `ns-a`.
+
+Warning: this will allow users with access to `generate-ssh-keys-task` and `create-vm-from-manifest-task` service accounts to run pods in any namespace.
+This will elevate their privileges to cluster admin level.
+
+
+# Running the tasks in new namespaces
```

Comment:
is "running" as task different from "deploying" it? If not, can you use the same verb? it seems that `new-ns` behaves like `ns-a`, but I'm not sure.

Maybe you can use `task-ns1` and `vm-ns1` and then add  `task-ns2` and `vm-ns2` so it's clearer for a layperson?

Reply:
That is indeed clearer. Changed it to use these namespace. Although `vm-ns2` was not necessary.

Reply:
But it is necessary to me: I'm trying to understand how one piplinerun can access resources from two different namespaces

Reply:
Hm, in that case I could add 2nd loop for the namespace to https://github.com/suomiy/kubevirt-tekton-tasks/blob/rbacUpdate/docs/tasks-rbac-permissions.md#multi-namespace-rbac-permissions example.

Although that might be too much for a simple use case regarding the readability?

Reply:
I would drop `task-ns2` for brevity. It's somehow more obvious to me to me that I can deploy the task wherever I want.

Reply:
- I added two namespaces to [#multi-namespace-rbac-permissions](https://github.com/suomiy/kubevirt-tekton-tasks/blob/rbacUpdate/docs/tasks-rbac-permissions.md#multi-namespace-rbac-permissions) example. Please let me know if the readability suffers.
- I think It is good to keep the `task-ns2` because there has to some namespace specified in [#deploying-the-tasks-in-additional-namespaces](https://github.com/suomiy/kubevirt-tekton-tasks/blob/rbacUpdate/docs/tasks-rbac-permissions.md#deploying-the-tasks-in-additional-namespaces) section.

Reply:
> I think It is good to keep the task-ns2 because there has to be some namespace specified in #deploying-the-tasks-in-additional-namespaces section.

My wrong, I guess there doesn't have to be - I suppose we could remove it. 

Not sure what is better though - verbosity(more clear) or reusability.

Reply:
Sorry, I've lost you here. Could you point me to the current description of multi-namespace usage, or that follow-up PR?